apps/api/src/app/workflows-v2/usecases/validate-content/prepare-and-validate-content/prepare-and-validate-content.usecase.ts

```diff
@@ -220,4 +243,70 @@ export class PrepareAndValidateContentUsecase {
 
     return result;
   }
+
+  private async computeTierIssues(
+    defaultControlValues: Record<string, unknown>,
+    user: UserSessionData,
+    stepType?: StepTypeEnum
+  ): Promise<Record<string, ContentIssue[]>> {
+    const deferDuration =
+      isValidDigestUnit(defaultControlValues.unit) && isNumber(defaultControlValues.amount)
+        ? calculateMilliseconds(defaultControlValues.amount, defaultControlValues.unit)
+        : 0;
+
+    const restrictionsErrors = await this.tierRestrictionsValidatorUsecase.execute({
+      deferDurationMs: deferDuration,
+      organizationId: user.organizationId,
+      stepType,
+    });
+
+    if (!restrictionsErrors) {
+      return {};
+    }
+
+    const result: Record<string, ContentIssue[]> = {};
+    for (const restrictionsError of restrictionsErrors) {
+      result.amount = [
+        {
+          issueType: StepContentIssueEnum.TIER_LIMIT_EXCEEDED,
+          message: restrictionsError.message,
+        },
+      ];
+      result.unit = [
+        {
+          issueType: StepContentIssueEnum.TIER_LIMIT_EXCEEDED,
+          message: restrictionsError.message,
+        },
+      ];
+    }
+
+    return result;
+  }
+}
+
+function calculateMilliseconds(amount: number, unit: DigestUnitEnum): number {
+  switch (unit) {
+    case DigestUnitEnum.SECONDS:
+      return amount * 1000;
+    case DigestUnitEnum.MINUTES:
+      return amount * 1000 * 60;
+    case DigestUnitEnum.HOURS:
+      return amount * 1000 * 60 * 60;
+    case DigestUnitEnum.DAYS:
+      return amount * 1000 * 60 * 60 * 24;
+    case DigestUnitEnum.WEEKS:
+      return amount * 1000 * 60 * 60 * 24 * 7;
+    case DigestUnitEnum.MONTHS:
+      return amount * 1000 * 60 * 60 * 24 * 30; // Using 30 days as an approximation for a month
+    default:
+      return 0;
+  }
+}
+
+function isValidDigestUnit(unit: unknown): unit is DigestUnitEnum {
+  return Object.values(DigestUnitEnum).includes(unit as DigestUnitEnum);
+}
+
+function isNumber(value: unknown): value is number {
+  return !Number.isNaN(Number(value));
```

Comment:
would expect it inside the use case

Reply:
not at the current use case structure. at the moment the use case expects the computed amount, and then it applies the product restriction on top of the data.

Reply:
can't we change it? there is a worker method computing this also, lets merge this logic

Reply:
as i said worker logic is more complex and fragile i would not touch it as part of this work.